plugins/dell-k2/fu-dell-k2-dpmux.c

```diff
@@ -98,6 +98,9 @@ fu_dell_k2_dpmux_write(FuDevice *device,
 						(gsize)fu_chunk_array_length(chunks));
 	}
 
+	/* appease daemon as a device reset is required */
+	fu_device_set_version(device, fu_firmware_get_version(firmware)); /* nocheck:set-version */
```

Comment:
ohh no, please don't do this -- the device version is supposed to be read from the device itself rather than taken "on trust" from the firmware.

Reply:
The device version won't get updated until the dock cable is being disconnected from the host, in this case the version check will fail anyway.

Reply:
"device version not updated on success, %s != %s" is the mentioned error message, unless inappropriate (for this device type) flags are added, e.g.: reboot/shutdown/activation.

https://github.com/fwupd/fwupd/blob/19177bdc66d6fc8d1bc6b99767f488da9d820414/src/fu-engine.c#L2150

Reply:
I recall we did something similar with dell-dock for WD19/WD22 for the same reason.  Maybe we want a private device flag for this behavior.

Reply:
sounds good idea, added `FWUPD_DEVICE_FLAG_INSTALL_SKIP_VERSION_CHECK`.

Reply:
> We need to set the WAIT_FOR_REPLUG flag in this case, 

We don't request users to take an immediate action, basically the devices should still at work without interrupts to normal user experience. The update should take effect when the cables out when it happens.

Reply:
Yeah; this device sets `FWUPD_DEVICE_FLAG_USABLE_DURING_UPDATE ` so that it can finish the update later on when the user unplugs or manually runs `fwupdmgr activate`